app/src/main/java/com/example/hris/ui/viewmodels/AddTimeLogsViewModel.kt

```diff
@@ -18,21 +17,20 @@ class AddTimeLogsViewModel @Inject constructor(
 ) : AndroidViewModel(application) {
 
     val loadingDialogState = MutableLiveData<Boolean>()
-    val callValue = MutableLiveData<ResponseModel>()
+    val callValue = hrisRepository.addTimeLogsResponse
 
     fun addTimeLogs(type: String) {
         viewModelScope.launch {
-            callApi(type)
+            loadingDialogState.value = true
+            hrisRepository.addTimeLogs(
+                type
+            )
+            loadingDialogState.value = false
         }
     }
 
-    private suspend fun callApi(type: String) {
-        loadingDialogState.value = true
-        val call = HrisApi.retrofitService.addTimeLogs(
-            hrisRepository.profileData.value!!.userID,
-            type
-        )
-        callValue.value = call
-        loadingDialogState.value = false
+    fun resetValue(){
+        hrisRepository.addTimeLogsResponse.value = null
```

Comment:
Instead of resetting the value of `hrisRepository.addTimeLogsResponse` every time, why not just remove caching of `AddTimeLogsResponse` inside the `HrisRepository` and just call `addTimeLogs()` every time you need its response?

Reply:
@davidimbong `LoginViewModel` and `UpdateProfileViewModel` still directly calls `HrisApi.retrofitService`